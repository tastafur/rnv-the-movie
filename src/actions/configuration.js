export const FETCH_CONFIGURATION = 'FETCH_CONFIGURATION';
export const SET_CONFIGURATION = 'SET_CONFIGURATION';

export function fetchConfiguration() {
  return {
    type: FETCH_CONFIGURATION,
  };
}

export function setConfiguration(configuration) {
  return {
    type: SET_CONFIGURATION,
    payload: {
      configuration
    }
  };
}
