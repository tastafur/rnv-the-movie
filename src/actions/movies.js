export const FETCH_MOVIES = 'FETCH_MOVIES';
export const SET_MOVIES = 'SET_MOVIES';

export function fetchMovies() {
  return {
    type: FETCH_MOVIES,
  };
}

export function setMovies(movies) {
  return {
    type: SET_MOVIES,
    payload: {
      movies
    }
  };
}
