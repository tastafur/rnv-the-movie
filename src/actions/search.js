export const FETCH_SEARCH = 'FETCH_SEARCH';
export const SET_SEARCH = 'SET_SEARCH';

export function fetchSearch(query) {
  return {
    type: FETCH_SEARCH,
    payload: {
      query
    }
  };
}

export function setSearch(entertainments) {
  return {
    type: SET_SEARCH,
    payload: {
      entertainments
    }
  };
}
