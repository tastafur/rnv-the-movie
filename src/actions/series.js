export const FETCH_SERIES = 'FETCH_SERIES';
export const SET_SERIES = 'SET_SERIES';

export function fetchSeries() {
  return {
    type: FETCH_SERIES,
  };
}

export function setSeries(series) {
  return {
    type: SET_SERIES,
    payload: {
      series
    }
  };
}
