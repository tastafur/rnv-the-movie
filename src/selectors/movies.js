import { getFieldsMainEntertainment } from './commons';
export const getMovieCover = movies =>
  movies.map(movie => getFieldsMainEntertainment(movie));
