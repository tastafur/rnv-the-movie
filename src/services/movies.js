import apiFetch from './apiFetch';

export const getMovies = (data = {}) =>
  apiFetch({
    endPoint: encodeURI('/discover/movie'),
    method: 'GET',
    params: data.params
  });
