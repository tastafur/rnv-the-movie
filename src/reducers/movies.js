import { SET_MOVIES } from '../actions/movies'

export const initialMovies = [];
// Here I could normalize the data so that it is easier for me to update one of the elements or delete it, for me it is a common use in other projects but I don't see it here
// https://redux.js.org/recipes/structuring-reducers/normalizing-state-shape/
// https://redux.js.org/recipes/structuring-reducers/updating-normalized-data
export function movies(state = initialMovies, { type, payload }) {
  switch (type) {
    case SET_MOVIES:
      return [
        ...state,
        ...payload.movies
      ];
    default:
      return state;
  }
}
