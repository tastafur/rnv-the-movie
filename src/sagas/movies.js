import { call, put, takeLatest } from 'redux-saga/effects';

// Services
import { getMovies } from '../services/movies';

// Actions
import { setMovies, FETCH_MOVIES } from '../actions/movies';

// utils errors
import { errorHandler } from './util/errorHandler'

export function* fetchMoviesProcess() {
  try {
    const resultMovies = yield call(getMovies, { params: { sort_by: 'popularity.desc' } });
    if(resultMovies) {
      yield put(setMovies(resultMovies.results))
    }
  } catch (e) {
    yield call(errorHandler, e)
  }
}

export function* watchFetchMoviesProcess() {
  yield takeLatest(FETCH_MOVIES, fetchMoviesProcess);
}
