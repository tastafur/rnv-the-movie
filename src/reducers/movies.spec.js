import { setMovies } from '../actions/movies'
import { movies as moviesReducer } from './movies';
import mocks from '../../mocks/data.json'

describe('Reducers', () => {
  describe('Movies', () => {
    it('should handle initial state', () => {
      const initialState = [];
      expect(moviesReducer(initialState, [])).toMatchSnapshot()
    });

    it('should set MOVIES', () => {
      const { movies } = mocks;
      const currentState = {};

      expect(moviesReducer(currentState, setMovies(movies))).toMatchSnapshot()
    })
  })
});
