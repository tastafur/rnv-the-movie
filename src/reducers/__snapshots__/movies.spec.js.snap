// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducers Movies should handle initial state 1`] = `Array []`;

exports[`Reducers Movies should set MOVIES 1`] = `
Object {
  "10911": Object {
    "adult": false,
    "backdrop_path": null,
    "genre_ids": Array [
      28,
      18,
      12,
      10752,
    ],
    "id": 10911,
    "original_language": "en",
    "original_title": "The Guns of Navarone",
    "overview": "A team of allied saboteurs are assigned an impossible mission: infiltrate an impregnable Nazi-held island and destroy the two enormous long-range field guns that prevent the rescue of 2,000 trapped British soldiers.",
    "popularity": 1.075583,
    "poster_path": null,
    "release_date": "1961-04-27",
    "title": "The Guns of Navarone",
    "video": false,
    "vote_average": 7.56,
    "vote_count": 50,
  },
  "164558": Object {
    "adult": false,
    "backdrop_path": null,
    "genre_ids": Array [
      99,
      10402,
    ],
    "id": 164558,
    "original_language": "en",
    "original_title": "One Direction: This Is Us",
    "overview": "Go behind the scenes during One Directions sell out \\"Take Me Home\\" tour and experience life on the road.",
    "popularity": 1.166982,
    "poster_path": null,
    "release_date": "2013-08-30",
    "title": "One Direction: This Is Us",
    "video": false,
    "vote_average": 8.45,
    "vote_count": 55,
  },
  "654": Object {
    "adult": false,
    "backdrop_path": null,
    "genre_ids": Array [
      80,
      18,
    ],
    "id": 654,
    "original_language": "en",
    "original_title": "On the Waterfront",
    "overview": "",
    "popularity": 1.07031,
    "poster_path": null,
    "release_date": "1954-06-22",
    "title": "On the Waterfront",
    "video": false,
    "vote_average": 8.19,
    "vote_count": 51,
  },
}
`;
